### ISSUE TITLE
Scylla hangs when it doesn't have permissions to TLS files

### ISSUE THREAD

**Issue by zimnx** [scylladb/scylladb#10305]
Scylla fails to start at late stage without any error message when it doesn't have permissions to read TLS cert and key needed for enabling encrypted native transport.

Relevant Scylla config:
```
client_encryption_options:
  enabled: true
  certificate: /etc/scylla/db.crt
  keyfile: /etc/scylla/db.key
  truststore: /etc/scylla/ca.crt
  require_client_auth: true
```
File permissions:
```
[root@ip-172-31-33-136 ec2-user]# ls -la /etc/scylla/
total 52
drwxr-xr-x  3 root root   119 03-30 19:58 .
drwxr-xr-x 86 root root  8192 03-30 19:18 ..
-rw-r--r--  1 root root  1915 03-30 19:57 ca.crt
drwxr-xr-x  2 root root   239 03-30 19:18 cassandra
-rw-r--r--  1 root root   366 03-06 12:54 cassandra-rackdc.properties
-rw-r--r--  1 root root  1915 03-30 19:57 db.crt
-rw-------  1 root root  3243 03-30 19:57 db.key
-rw-r--r--  1 root root 23566 03-30 19:58 scylla.yaml
```

Scylla journal:

```
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal systemd[1]: Starting Scylla Server...
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: Scylla version 4.6.0-0.20220306.2f2d22a86 with build-id d487cdea729e95d8caebe2e9e2ae8698f35f19ee starting ...
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: command used: "/usr/bin/scylla --log-to-syslog 0 --log-to-stdout 1 --default-log-level info --network-stack posix --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 1-2,7-8 --smp 4 --memory=33710M"
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: parsed command line options: [log-to-syslog, (positional) 0, log-to-stdout, (positional) 1, default-log-level, (positional) info, network-stack, (positional) posix, io-properties-file: /etc/scylla.d/io_properties.yaml, cpuset, (positional) 1-2,7-8, smp, (positional) 4, memory: 33710M]
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: WARN  2022-03-30 20:10:48,822 [shard 0] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: WARN  2022-03-30 20:10:48,863 [shard 1] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: WARN  2022-03-30 20:10:48,886 [shard 3] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: WARN  2022-03-30 20:10:48,886 [shard 2] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,889 [shard 0] seastar - updated: blocked-reactor-notify-ms=1000000
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,889 [shard 1] seastar - updated: blocked-reactor-notify-ms=1000000
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,889 [shard 2] seastar - updated: blocked-reactor-notify-ms=1000000
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,889 [shard 3] seastar - updated: blocked-reactor-notify-ms=1000000
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,893 [shard 0] init - installing SIGHUP handler
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,916 [shard 0] init - Scylla version 4.6.0-0.20220306.2f2d22a86 with build-id d487cdea729e95d8caebe2e9e2ae8698f35f19ee starting ...
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,916 [shard 0] init - starting prometheus API server
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,917 [shard 0] init - starting tokens manager
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,917 [shard 0] init - starting migration manager notifier
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,917 [shard 0] init - starting lifecycle notifier
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,917 [shard 0] init - creating tracing
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,917 [shard 0] init - creating snitch
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,917 [shard 0] init - determining DNS name
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,917 [shard 0] init - starting API server
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,917 [shard 0] init - Scylla API server listening on 127.0.0.1:10000 ...
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,919 [shard 0] service_level_controller - update_from_distributed_data: starting configuration polling loop
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,919 [shard 0] init - starting gossiper
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,919 [shard 0] init - seeds={172.31.33.136}, listen_address=172.31.33.136, broadcast_address=172.31.33.136
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,919 [shard 0] init - starting Raft service
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,919 [shard 0] init - initializing storage service
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,919 [shard 0] storage_service - Started node_ops_abort_thread
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,919 [shard 1] storage_service - Started node_ops_abort_thread
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,919 [shard 3] storage_service - Started node_ops_abort_thread
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,919 [shard 2] storage_service - Started node_ops_abort_thread
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,919 [shard 0] init - starting per-shard database core
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,919 [shard 0] init - creating and verifying directories
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,938 [shard 0] init - starting database
mar 30 20:10:48 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:48,959 [shard 0] database - Populating Keyspace system_schema
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,045 [shard 0] seastar - updated: blocked-reactor-notify-ms=200
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,045 [shard 1] seastar - updated: blocked-reactor-notify-ms=200
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,045 [shard 3] seastar - updated: blocked-reactor-notify-ms=200
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,045 [shard 2] seastar - updated: blocked-reactor-notify-ms=200
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,045 [shard 0] init - starting storage proxy
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,051 [shard 0] init - starting migration manager
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,051 [shard 0] init - starting query processor
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,052 [shard 0] init - initializing batchlog manager
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,053 [shard 0] format_selector - Selected md sstables format
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,053 [shard 0] legacy_schema_migrator - Moving 0 keyspaces from legacy schema tables to the new schema keyspace (system_schema)
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,054 [shard 0] legacy_schema_migrator - Dropping legacy schema tables
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,067 [shard 0] legacy_schema_migrator - Completed migration of legacy schema tables
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,067 [shard 0] init - loading system sstables
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,072 [shard 0] init - loading non-system sstables
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,106 [shard 0] database - Populating Keyspace system_traces
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,106 [shard 0] database - Keyspace system_traces: Reading CF sessions_time_idx id=0ebf001c-c1d1-3693-9a63-c3d96ac53318 version=d80bfb8b-bb07-34c5-8e50-80395de23041
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,106 [shard 0] database - Keyspace system_traces: Reading CF sessions id=c5e99f16-8677-3914-b17e-960613512345 version=4682ce3d-17fb-3979-8563-6203b096722c
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,106 [shard 0] database - Keyspace system_traces: Reading CF node_slow_log_time_idx id=f9706768-aa1e-3d87-9e5c-51a3927c2870 version=5038e46f-d51e-3c86-a985-d5044b1e1e34
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,106 [shard 0] database - Keyspace system_traces: Reading CF node_slow_log id=bfcc4e62-5b63-3aa1-a1c3-6f5e47f3325c version=24b2f10f-3309-3e6d-945d-7bd5b8b82944
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,106 [shard 0] database - Keyspace system_traces: Reading CF events id=8826e8e9-e16a-3728-8753-3bc1fc713c25 version=2a3150ee-39a2-3de5-a4d1-137ec32c5471
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,106 [shard 0] database - Populating Keyspace system_auth
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,106 [shard 0] database - Keyspace system_auth: Reading CF roles id=5bc52802-de25-35ed-aeab-188eecebb090 version=7a6b9a7b-2b0d-3afd-90f0-d9a4609844f4
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,106 [shard 0] database - Keyspace system_auth: Reading CF role_permissions id=3afbe79f-2194-31a7-add7-f5ab90d8ec9c version=909d45a0-6e3f-30cf-9cc9-dc9670eecb9f
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,106 [shard 0] database - Keyspace system_auth: Reading CF role_members id=0ecdaa87-f8fb-3e60-88d1-74fb36fe5c0d version=ba73c274-5709-3c37-b997-953fb97523ce
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,107 [shard 0] database - Keyspace system_auth: Reading CF role_attributes id=6b8c7359-a843-33f2-a1d8-5dc6a187436f version=db323a72-29cb-3779-8a33-f8372c385233
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,107 [shard 0] database - Populating Keyspace system_distributed_everywhere
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,107 [shard 0] database - Keyspace system_distributed_everywhere: Reading CF cdc_generation_descriptions_v2 id=234d2227-dd63-3d37-ac5f-c013e2ea9e6e version=f04f0def-50da-31c9-96b1-ab22ba0cc08e
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,107 [shard 0] database - Populating Keyspace system_distributed
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,107 [shard 0] database - Keyspace system_distributed: Reading CF view_build_status id=5582b59f-8e4e-35e1-b913-3acada51eb04 version=e61ea49b-8dbc-3ff6-a69f-896f4d38b7bb
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,107 [shard 0] database - Keyspace system_distributed: Reading CF service_levels id=b8c556bd-212d-37ad-9484-690c73a5994b version=8a951d7a-1758-346d-953f-b8cc9cbc0f96
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,107 [shard 0] database - Keyspace system_distributed: Reading CF cdc_streams_descriptions_v2 id=0bf73fd7-65b2-36b0-85e5-658131d5df36 version=58a18c32-0c5a-34ba-992f-7723dd0265c0
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,107 [shard 0] database - Keyspace system_distributed: Reading CF cdc_generation_timestamps id=fdf455c4-cfec-3e00-9719-d7a45436c89d version=5777e043-a81a-3324-8803-b497c056849f
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,119 [shard 0] init - starting view update generator
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,119 [shard 0] init - setting up system keyspace
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,131 [shard 0] init - starting commit log
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,133 [shard 0] init - initializing migration manager RPC verbs
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,133 [shard 0] init - initializing storage proxy RPC verbs
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,133 [shard 0] init - starting streaming service
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,133 [shard 0] init - starting hinted handoff manager
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,135 [shard 0] init - starting messaging service
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,135 [shard 0] repair - Setting max_repair_memory=883530137, max_repair_memory_per_range=33554432, max_repair_ranges_in_parallel=6
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,135 [shard 0] init - starting CDC Generation Management service
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,135 [shard 0] init - starting CDC log service
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,135 [shard 0] init - starting storage service
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal systemd[1]: Started Scylla Server.
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,136 [shard 0] init - starting sstables loader
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,136 [shard 0] messaging_service - Starting Messaging Service on port 7000
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,136 [shard 0] storage_service - Loading persisted ring state
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,140 [shard 0] storage_service - initial_contact_nodes={172.31.33.136}, loaded_endpoints={}, loaded_peer_features=0
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,140 [shard 0] storage_service - Checking remote features with gossip, initial_contact_nodes={172.31.33.136}
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,140 [shard 0] gossip - Gossip shadow round started with nodes={}
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: WARN  2022-03-30 20:10:49,140 [shard 0] gossip - All nodes={} are down for get_endpoint_states verb. Skip ShadowRound.
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,140 [shard 0] gossip - Gossip shadow round finisehd with nodes_talked={}
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,140 [shard 0] gossip - Feature check passed. Local node 172.31.33.136 features = {CDC, CDC_GENERATIONS_V2, COMPUTED_COLUMNS, CORRECT_COUNTER_ORDER, CORRECT_IDX_TOKEN_IN_SECONDARY_INDEX, CORRECT_NON_COMPOUND_RANGE_TOMBSTONES, CORRECT_STATIC_COMPACT_IN_MC, COUNTERS, DIGEST_FOR_NULL_VALUES, DIGEST_INSENSITIVE_TO_EXPIRY, DIGEST_MULTIPARTITION_READ, HINTED_HANDOFF_SEPARATE_CONNECTION, INDEXES, LARGE_PARTITIONS, LA_SSTABLE_FORMAT, LWT, MATERIALIZED_VIEWS, MC_SSTABLE_FORMAT, MD_SSTABLE_FORMAT, NONFROZEN_UDTS, PER_TABLE_CACHING, PER_TABLE_PARTITIONERS, RANGE_SCAN_DATA_VARIANT, RANGE_TOMBSTONES, ROLES, ROW_LEVEL_REPAIR, SCHEMA_TABLES_V3, STREAM_WITH_RPC_STREAM, TRUNCATION_TABLE, UDA, UNBOUNDED_RANGE_TOMBSTONES, VIEW_VIRTUAL_COLUMNS, WRITE_FAILURE_REPLY, XXHASH}, Remote common_features = {}
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,142 [shard 0] storage_service - Restarting a node in NORMAL status
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,171 [shard 0] schema_tables - Schema version changed to 617833bb-62e0-3f2d-8890-9c5be0500fa0
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,171 [shard 0] storage_service - Starting up server gossip
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,174 [shard 0] compaction - [Compact system.compaction_history 7ddd18b0-b065-11ec-8314-f508921f074f] Compacted 2 sstables to [/var/lib/scylla/data/system/compaction_history-b4dbb7b4dc493fb5b3bfce6e434832ca/md-20-big-Data.db:level=0]. 136kB to 88kB (~65% of original) in 25ms = 3MB/s. ~256 total partitions merged to 22.
mar 30 20:10:49 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:10:49,176 [shard 0] gossip - failure_detector_loop: Started main loop
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] gossip - No gossip backlog; proceeding
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature CDC is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature CDC_GENERATIONS_V2 is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature COMPUTED_COLUMNS is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature CORRECT_IDX_TOKEN_IN_SECONDARY_INDEX is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature DIGEST_FOR_NULL_VALUES is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature DIGEST_INSENSITIVE_TO_EXPIRY is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature HINTED_HANDOFF_SEPARATE_CONNECTION is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature LWT is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature MD_SSTABLE_FORMAT is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature NONFROZEN_UDTS is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature PER_TABLE_CACHING is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature PER_TABLE_PARTITIONERS is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature RANGE_SCAN_DATA_VARIANT is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature UDA is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,177 [shard 0] features - Feature VIEW_VIRTUAL_COLUMNS is enabled
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,179 [shard 0] init - starting system distributed keyspace
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,182 [shard 0] schema_tables - Schema version changed to 0838d873-6c47-3e09-9ab5-f691077d828b
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,185 [shard 0] schema_tables - Schema version changed to 0838d873-6c47-3e09-9ab5-f691077d828b
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,188 [shard 0] schema_tables - Schema version changed to 0838d873-6c47-3e09-9ab5-f691077d828b
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,191 [shard 0] schema_tables - Schema version changed to 0838d873-6c47-3e09-9ab5-f691077d828b
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,193 [shard 0] schema_tables - Schema version changed to 0838d873-6c47-3e09-9ab5-f691077d828b
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,223 [shard 0] storage_service - NORMAL: node is now in normal status
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,223 [shard 0] cdc - Latest generation seen during startup: (2022/03/30 20:03:31, 0347f492-aaec-428e-805f-b70dfaaf2aab)
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,225 [shard 0] cdc - Starting to use generation (2022/03/30 20:03:31, 0347f492-aaec-428e-805f-b70dfaaf2aab)
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,225 [shard 0] cdc - Generation (2022/03/30 20:03:31, 0347f492-aaec-428e-805f-b70dfaaf2aab): streams description table already updated.
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,225 [shard 0] init - starting tracing
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,226 [shard 0] init - SSTable data integrity checker is disabled.
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,226 [shard 0] init - starting auth service
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,227 [shard 0] init - starting batchlog manager
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,227 [shard 0] init - starting load meter
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,227 [shard 0] init - starting cf cache hit rate calculator
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,227 [shard 0] init - starting view update backlog broker
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,227 [shard 0] init - Waiting for gossip to settle before accepting client requests...
mar 30 20:11:01 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:01,228 [shard 3] compaction - [Compact system.peers 850fbf70-b065-11ec-a572-f50a921f074f] Compacted 2 sstables to [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/md-31-big-Data.db:level=0]. 11kB to 5kB (~50% of original) in 3ms = 1MB/s. ~256 total partitions merged to 1.
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,227 [shard 0] gossip - No gossip backlog; proceeding
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,227 [shard 0] init - allow replaying hints
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,228 [shard 0] init - Launching generate_mv_updates for non system tables
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,228 [shard 0] init - starting the view builder
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,232 [shard 0] init - starting native transport
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 0] init - Shutting down view builder API
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 0] init - Shutting down view builder API was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 0] init - Shutting down view builder
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 0] view - Stopping view builder
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 0] view - Draining view builder
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 1] view - Stopping view builder
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 1] view - Draining view builder
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 3] view - Stopping view builder
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 2] view - Stopping view builder
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 3] view - Draining view builder
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 2] view - Draining view builder
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 0] init - Shutting down view builder was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 0] init - Shutting down hinted handoff API
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 0] init - Shutting down hinted handoff API was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,233 [shard 0] init - Shutting down view update backlog broker
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down view update backlog broker was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down cf cache hit rate calculator
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down cf cache hit rate calculator was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down load meter
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down load meter was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down batchlog manager
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down batchlog manager was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down snapshots API
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down snapshots API was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down snapshots
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down snapshots was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down auth service
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down auth service was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down tracing
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] tracing - Asked to shut down
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] tracing - Tracing is down
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 1] tracing - Asked to shut down
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 3] tracing - Asked to shut down
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 2] tracing - Asked to shut down
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 2] tracing - Tracing is down
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 3] tracing - Tracing is down
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 1] tracing - Tracing is down
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down tracing was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down storage service notifications
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down storage service notifications was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down system distributed keyspace
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down system distributed keyspace was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down storage service notifications
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down storage service notifications was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down sstables loader API
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down sstables loader API was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down sstables loader
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down sstables loader was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down repair API
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down repair API was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down messaging service API
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down messaging service API was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down storage service messaging
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down storage service messaging was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down cdc log service
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down cdc log service was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,234 [shard 0] init - Shutting down CDC Generation Management service
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down CDC Generation Management service was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down repair service
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down repair service was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down drain storage proxy
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] hints_manager - Asked to stop
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] hints_manager - Stopped
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] hints_manager - Asked to stop
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] hints_manager - Stopped
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 1] hints_manager - Asked to stop
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 2] hints_manager - Asked to stop
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 3] hints_manager - Asked to stop
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 2] hints_manager - Stopped
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 2] hints_manager - Asked to stop
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 1] hints_manager - Stopped
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 2] hints_manager - Stopped
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 1] hints_manager - Asked to stop
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 1] hints_manager - Stopped
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 3] hints_manager - Stopped
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 3] hints_manager - Asked to stop
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 3] hints_manager - Stopped
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down drain storage proxy was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down streaming service
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down streaming service was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down storage proxy RPC verbs
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down storage proxy RPC verbs was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down sstables format selector
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down sstables format selector was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down migration manager
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] migration_manager - stopping migration service
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 1] migration_manager - stopping migration service
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 3] migration_manager - stopping migration service
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 2] migration_manager - stopping migration service
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down migration manager was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] init - Shutting down database
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] compaction_manager - Asked to stop
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 0] compaction_manager - Stopping 1 ongoing compactions due to shutdown
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 1] compaction_manager - Asked to stop
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 3] compaction_manager - Asked to stop
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 2] compaction_manager - Asked to stop
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 1] compaction_manager - Stopping 0 ongoing compactions due to shutdown
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 3] compaction_manager - Stopping 0 ongoing compactions due to shutdown
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 1] compaction_manager - Stopped
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 3] compaction_manager - Stopped
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 2] compaction_manager - Stopping 0 ongoing compactions due to shutdown
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,235 [shard 2] compaction_manager - Stopped
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,238 [shard 0] compaction - [Compact system.truncated 8c382d00-b065-11ec-8314-f508921f074f] Compacted 2 sstables to [/var/lib/scylla/data/system/truncated-38c19fd0fb863310a4b70d0cc66628aa/md-68-big-Data.db:level=0]. 82kB to 76kB (~93% of original) in 5ms = 15MB/s. ~256 total partitions merged to 2.
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,238 [shard 0] compaction_manager - Stopped
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,316 [shard 0] init - Shutting down database was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,316 [shard 0] init - Shutting down service_memory_limiter
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,316 [shard 0] init - Shutting down service_memory_limiter was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,316 [shard 0] init - Shutting down sst_dir_semaphore
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,316 [shard 0] init - Shutting down sst_dir_semaphore was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,316 [shard 0] init - Shutting down storage_service
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,316 [shard 0] init - Shutting down storage_service was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,317 [shard 0] init - Shutting down Raft
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,317 [shard 0] init - Shutting down Raft was successful
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,317 [shard 0] init - Shutting down gossiper
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,317 [shard 0] gossip - My status = NORMAL
mar 30 20:11:13 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:13,317 [shard 0] gossip - Announcing shutdown
mar 30 20:11:15 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:15,317 [shard 0] gossip - Disable and wait for gossip loop started
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:16,177 [shard 0] gossip - Gossip is now stopped
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:16,177 [shard 0] init - Shutting down gossiper was successful
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:16,177 [shard 0] init - Shutting down messaging service
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:16,177 [shard 0] init - Shutting down messaging service was successful
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:16,177 [shard 0] init - Shutting down service level controller
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:16,178 [shard 0] service_level_controller - update_from_distributed_data: configuration polling loop aborted
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:16,178 [shard 0] init - Shutting down service level controller was successful
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:16,178 [shard 0] init - Shutting down API server
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:16,178 [shard 0] init - Shutting down API server was successful
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal scylla[25943]: INFO  2022-03-30 20:11:16,178 [shard 0] init - Shutting down tracing instance
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal systemd[1]: scylla-server.service: main process exited, code=exited, status=1/FAILURE
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal systemd[1]: Unit scylla-server.service entered failed state.
mar 30 20:11:16 ip-172-31-33-136.eu-west-1.compute.internal systemd[1]: scylla-server.service failed.

```

After fixing permissions:
```
[root@ip-172-31-33-136 ec2-user]# chmod 644 /etc/scylla/db.key
[root@ip-172-31-33-136 ec2-user]# ls -la /etc/scylla/
total 52
drwxr-xr-x  3 root root   119 03-30 19:58 .
drwxr-xr-x 86 root root  8192 03-30 19:18 ..
-rw-r--r--  1 root root  1915 03-30 19:57 ca.crt
drwxr-xr-x  2 root root   239 03-30 19:18 cassandra
-rw-r--r--  1 root root   366 03-06 12:54 cassandra-rackdc.properties
-rw-r--r--  1 root root  1915 03-30 19:57 db.crt
-rw-r--r--  1 root root  3243 03-30 19:57 db.key
-rw-r--r--  1 root root 23566 03-30 19:58 scylla.yaml
```

Scylla starts fine.

*Installation details*
Scylla version (or git commit hash): 4.6.0
OS (RHEL/CentOS/Ubuntu/AWS AMI): Amazon Linux, Scylla installed from RHEL repo

*Hardware details (for performance issues)*    
Platform (physical/VM/cloud instance type/docker): VM

**Comment by elcallio** [comment]
I ran it manually - I can barely use systemd. :-)
But as long as things are in a log message, the eventual collection thereof should be the same, no? Unless scylla actually outputs directly to systemd (?) and this error message is _after_ it somehow detached? Sounds far fetched, but that would be all exit errors being dropped if so.

**Comment by bhalevy** [comment]
@lersek can you please try to reproduce this issue and fix it or close it if it was somehow taken care of?

**Comment by lersek** [comment]
After a few hours of debugging...

The symptom is actually more insidious than reported (with scylla master @ `607abe96e864`). Not only is the exception not reported, but scylla *gets stuck*, and you can't even cleanly shut it down by `^C` or `systemctl stop`.

Here's how far I've managed to get tonight.

~~~
scylla_main()                                         [main.cc]
  register_protocol_server(cql_server_ctl)            [service/storage_service.cc]
    start_server()                                    [transport/controller.cc]
      do_start_server()                               [transport/controller.cc]
        defer sharded<cql_server>::stop().get()
        defer unsubscribe_server().get()
        start_listening_on_tcp_sockets()              [transport/controller.cc]
          configure_tls_creds_builder()               [db/config.cc]
            credentials_builder::set_x509_key_file()  [seastar/src/net/tls.cc]
              read_fully()                            [seastar/src/net/tls.cc]
                open_file_dma()                       [seastar/src/core/reactor.cc]
                  throw_fs_exception_if_error()       [seastar/src/core/syscall_result.hh]
                    throw_fs_exception()              [seastar/src/core/syscall_result.hh]
                      throw fs::filesystem_error()
                throw_with_nested(std::runtime_error)
        ~ unsubscribe_server().get()
        ~ sharded<cql_server>::stop().get() --> HANGS
~~~

In summary:

When the key file / certificate file etc. cannot be opened, the original exception is an `fs::filesystem_error`. This is then nested into a `std::runtime_error`. This exception propagates up to `do_start_server()`, and when the exception tries to unwind *that* frame, the deferred cleanups run. The cleanup that was deferred last runs first (`unsubscribe_server().get()`), it completes fine. The cleanup that was deferred first runs last (`cserver->stop().get()`), and it never returns; it gets stuck.

In turn this means that `register_protocol_server(cql_server_ctl)` never returns to `scylla_main()`, and we never reach the part that says

~~~
            stop_signal.wait().get();
            startlog.info("Signal received; shutting down");
~~~

Therefore the server is in effect stuck during initialization, and cannot even be shut down cleanly. (`SIGKILL` or `SIGQUIT` is needed.)

The `defer` in question was introduced in commit `12220b74e874` ("transport/controller: Rework try-catch into defers", 2021-06-04). However, I don't think it's the culprit. Namely, that commit was first released in `scylla-4.6.0`, which is what this particular ticket was also reported against -- but the symptom back then was much milder. The original problem was that the exception was (perhaps) not properly logged (and the `defer` at hand didn't interfere with that); now we can't even test how the exception is ultimately caught and handled, because we hang earlier than that. :/

A bisection between 4.6.0 and current master should be a no-brainer (for locating the commit that introdued the hang); however, this is modern C++, where a single source file may compile for 90 seconds or more :/ Thus, analyzing the hang might be faster than bisection. I'll see.

Do we have an archive of RPMs somewhere? If we have historical binaries, that could narrow down the bisection. [This page](https://opensource.docs.scylladb.com/stable/getting-started/install-scylla/install-on-linux.html) claims that arbitrary versions can be installed from the DNF repo; however, I've tried in a Rocky-9.4 VM, and only the latest release seems available (`scylla-6.1.0-0.20240804.abbf0b24a60c.x86_64`).

**Comment by lersek** [comment]
> Next: investigate why and how exactly commit `6b178f9a4a90` broke the deferred cleanups.

That's not hard to see.

Before commit `6b178f9a4a90`, the `configure_tls_creds_builder()` function would be called *ahead of* registering the two deferred cleanup actions.

At commit `6b178f9a4a90`, the `configure_tls_creds_builder()` function call got moved inside new function `start_listening_on_tcp_sockets()`, and this new function `start_listening_on_tcp_sockets()` would be called *after* said deferrals.

Therefore, if `configure_tls_creds_builder()` were to throw an exception, then before commit `6b178f9a4a90`, the seastar thread scope created by `do_start_server()` would *not* include the deferred actions, but at commit `6b178f9a4a90`, it would.

This part:
~~~
         cserver->start(std::ref(_qp), std::ref(_auth_service), std::ref(_mem_limiter), std::move(get_cql_server_config), std::ref(cfg), std::ref(_sl_controller), std::ref(_gossiper), _cql_opcode_stats_key, _used_by_maintenance_socket).ge>
         auto on_error = defer([&cserver] { cserver->stop().get(); });
 
         subscribe_server(*cserver).get();
         auto on_error_unsub = defer([this, &cserver] {
             unsubscribe_server(*cserver).get();
         });
~~~
had been meant to execute only after the socket configuration was _nearly complete_, and only the `cql_server::listen()` call remained. Commit `6b178f9a4a90` broke that, by moving all of the socket setup after the quoted logic. I guess I'll have to disentangle that. :/

**Comment by lersek** [comment]
At the same time, I wonder why the deferred cleanups don't work, after commit `6b178f9a4a90`. In theory anyway, the deferred cleanups must mirror `cserver->start()` and `subscribe_server()`, and commit `6b178f9a4a90` changed nothing about that.

My hypothesis is that these deferred cleanups have never been actually exercised, before commit `6b178f9a4a90`. In other words, commit `6b178f9a4a90` may only have exposed a latent bug, but not caused the bug. Note that before commit `6b178f9a4a90`, there was precious little that could go wrong after these deferrals, so those parts would most likely never throw, and then at the end of the scope, the deferred cleanups would be canceled -- never actually put to use.

**Comment by bhalevy** [comment]
> Do we have an archive of RPMs somewhere? If we have historical binaries, that could narrow down the bisection.

@lersek see http://backtrace.scylladb.com/releases.html

**Comment by mykaul** [comment]
We also have a Jenkins job who can do a git bisect, IIRC.

**Comment by lersek** [comment]
The hang occurs regardless of deferral; if I do

~~~
        cserver->start(...).get();
        cserver->stop().get();
~~~

the `stop()` hangs.

**Comment by lersek** [comment]
`cserver->stop()` is in fact `server::stop()` [generic_server.cc].

It hangs for the following reason: `co_await shutdown()` is executed and it completes fine; however, at the end of the function, `co_await _all_connections_stopped.get_future()` hangs.

And *that* in turn hangs because there is never a call to `server::maybe_stop()`.

There are two ways to get `server::maybe_stop()` called:
1. A `connection` must exist and then it must be destroyed. The `connection` destructor in `generic_server.cc` calls `_server.maybe_stop()`. In our case, no connection is ever accepted, so this path is never taken.
2. `server::listen -> server::do_accepts -> maybe_stop()`. In our case, this never happens, because inside `start_listening_on_tcp_sockets()`, `configure_tls_creds_builder()` throws the exception before reaching `listen_on_all_shards()`, and hence `server.listen()`.

After commit `6b178f9a4a90` ("transport/controller: split configuring sockets into separate functions", 2024-02-05), we've got this:
- start CQL server
- defer stoppage
- attempt to set up client encryption options --> throw exception
- CQL server "listen" operation never reached --> `maybe_stop` never reached
- deferred stoppage gets activated
- coroutine deadlock because `maybe_stop` was never called.

Next: check the tree out again at the parent of `6b178f9a4a90`, and modify the code slightly: throw an exception after deferring both cleanups but before the first `cql_server::listen`. If we get a hang that way then stopping the CQL server right after starting it will have never been valid.

**Comment by lersek** [comment]
Indeed, calling `cserver->stop().get()` right after `cserver->start()` does hang just before commit `6b178f9a4a90`. Meaning that the bug predates commit `6b178f9a4a90`.

I don't know yet whether it's the client code that's invalid (invalid API use) or the `generic_server` class code (broken impl of exposed APIs).

**Comment by bhalevy** [comment]
@elcallio / @gleb-cloudius any insight on the above?

**Comment by elcallio** [comment]
Seems to me like `generic_server::shutdown` should call `maybe_stop` to guard against neither listen or connection having been created yet.

**Comment by lersek** [comment]
Bingo, the `cserver->stop().get()` call causes a coroutine deadlock right at its introduction in commit ae4d5a60ca61. That action has never worked, it's just that it had never been put to actual use until commit 6b178f9a4a90 (which would exercise it upon client encryption options failure).

That said, I still don't know whether it's a wrong API call, or a wrong API implementation.

**Comment by elcallio** [comment]
shutdown is broken. Again, needs changing as stated above. You cannot wait for a future that might not be set (no connections/no listeners).

**Comment by lersek** [comment]
@elcallio what is the purpose of the `_all_connections_stopped` fields? Thus far I can only tell it comes from commit 7b46c2da5388a (technically). The comment on `server::maybe_stop()` says
~~~
// Signal that all connections are stopped if the server is stopping and can be stopped.
~~~
and I can't make heads or tails of it.

My concern is that adding a maybe_stop() call to `server::shutdown()` might break something else.

Shutdown is documented like this in `generic_server.hh`:
~~~
    // Makes sure listening sockets no longer generate new connections and aborts the
    // connected sockets, so that new requests are not served and existing requests don't
    // send responses back.
    //
    // It does _not_ wait for any internal activity started by the established connections
    // to finish. It's the .stop() method that does it
    future<> shutdown();
~~~

I don't know if that matches the (somewhat incomprehensible) above-quoted comment on `server::maybe_stop()`.

**Comment by elcallio** [comment]
Reading the code, the method is somewhat misnamed, but all it does is really check the connection counters. If all are zero (no active connections) it sets the promise we eventually wait for in shutdown. I.e. calling it from shutdown is safe, because it does not itself modify any usage counters. However, you would have to either rewrite the promise, otherwise we can set it twice, or, maybe simpler, make shutdown() only wait for `_all_connections_stopped` if `_connections_being_accepted || _current_connections` - i.e. skip waiting for sync future if no connections are active.

**Comment by lersek** [comment]
Thanks, that makes sense!

**Comment by lersek** [comment]
I'll try to make that work, but still, an argument: I think that `_all_connections_stopped` being a future (i.e., something that is completable / settable only once) is not the greatest choice. If  `_all_connections_stopped` stands for "there's nothing to wait for to finish", then there are at least two transitions. Upon start, the predicate is "no need to wait (nothing exists yet)". The first transition is in listen ("yes, you need to wait"), and the second transition is when the last straggler goes away ("no need to wait more"). So it's more like a flip-flop than a single-shot promise.

**Comment by lersek** [comment]
Put differently, I feel the problem is that `_listeners_stopped` *being ready* can stand for two situations, when we reach
```
    co_await _all_connections_stopped.get_future();
```

The first situation (with `_listeners_stopped` being ready)  is that `_listeners_stopped` was explicitly populated with a future in `server::listen()`, and that future was awaited-for via `co_await shutdown()`. In that case, reaching `co_await _all_connections_stopped.get_future()` is safe.

The other situation is when `_listeners_stopped` is ready right from construction, in `generic_server.hh`:
~~~
    future<> _listeners_stopped = make_ready_future<>();
~~~

In this case, `co_await shutdown()` will succeed just the same, but the subsequent `co_await _all_connections_stopped.get_future()` will deadlock.

**Comment by elcallio** [comment]
You are not incorrect, the code should maybe use a gate instead, putting a gate holder in all connections.

**Comment by lersek** [comment]
Thank you for the idea! Conversion to `seastar::gate` and `seastar::gate::holder` seems to have solved the immediate issue; now I'll have to run the test suite.

**Comment by lersek** [comment]
OK. The unit tests have passed locally (in debug and release mode).

Regarding the original problems reported in this ticket, with the hang averted:
- When starting scylla locally, initialization takes a while, but it ultimately fails gracefully, with an error message such as

    `ERROR 2024-08-19 20:43:24,904 [shard 0:main] init - Startup failed: std::_Nested_exception<std::runtime_error> (Could not read certificate file conf/scylla.crt): std::filesystem::__cxx11::filesystem_error (error system:2, filesystem error: open failed: No such file or directory [conf/scylla.crt])`.

    That matches [this comment](https://github.com/scylladb/scylladb/issues/10305#issuecomment-1122710840).

- I've also built RPMs locally (`./configure --use-cmake && ninja -C build dist-{cqlsh,jmx,python3,server,tools}-rpm:RelWithDebInfo`), and used those to update the originally installed `scylla-6.1.0-0.20240804.abbf0b24a60c.x86_64` in my above-mentioned Rocky-9.4 VM. The RPMs are relevant because of [this systemd question](https://github.com/scylladb/scylladb/issues/10305#issuecomment-1123419089).

    After starting scylla with `systemctl start scylla-server`, and polling `systemctl status scylla-server`, the server's status is reported as `active (running)` for 1-2 seconds (and `systemctl status` displays the tail of the numerous initialization messages from the journal); but after ~2 seconds, the status correctly changes to `failed (Result: exit-code)`, and the lines quoted from the end of the journal contain the relevant exception, such as:

    `Aug 19 20:41:34 rocky9 scylla[2624]:  [shard 0:main] init - Startup failed: std::_Nested_exception<std::runtime_error> (Could not read key file /etc/scylla/rocky9.key): std::filesystem::__cxx11::filesystem_error (error system:13, filesystem error: open failed: Permission denied [/etc/scylla/rocky9.key])`

    I think this should be good enough. I first thought that with tighter systemd integration, we could expose a state like "initializing" before `active (running)`, but after some superficial web searching, that doesn't seem possible. There's one state called `activating`, but that's external to the service -- it's a [systemd state](https://unix.stackexchange.com/questions/771073/what-exactly-does-it-mean-for-a-systemd-service-to-be-activating).

I'll submit the `seastar::gate` conversion as a PR; let's hope that the dtest suite will pass too.

@zimnx can you please retitle this ticket to: "Scylla *hangs* when it doesn't have permissions to access TLS files" (emphasis mine; not needed in the title). Thanks.
